python/lancedb/embeddings/functions.py

```diff
@@ -28,25 +37,33 @@ class EmbeddingFunctionRegistry:
 
     @classmethod
     def get_instance(cls):
-        return REGISTRY
+        return __REGISTRY__
 
     def __init__(self):
         self._functions = {}
 
-    def register(self):
+    def register(self, alias: str = None):
         """
         This creates a decorator that can be used to register
-        an EmbeddingFunctionModel.
+        an EmbeddingModel.
+
+        Parameters
+        ----------
+        alias : Optional[str]
+            a human friendly name for the embedding function. If not
+            provided, the class name will be used.
         """
 
         # This is a decorator for a class that inherits from BaseModel
         # It adds the class to the registry
         def decorator(cls):
-            if not issubclass(cls, EmbeddingFunctionModel):
-                raise TypeError("Must be a subclass of EmbeddingFunctionModel")
+            if not issubclass(cls, EmbeddingFunction):
+                raise TypeError("Must be a subclass of EmbeddingModel")
             if cls.__name__ in self._functions:
                 raise KeyError(f"{cls.__name__} was already registered")
-            self._functions[cls.__name__] = cls
+            key = alias or cls.__name__
```

Comment:
Maybe it's better to check for `key` instead of `cls.__name__` in `self._functions` ?
I'm thinking of a case where user accidentally overrides an embedding function using an alias.

Reply:
yeah we lookup by alias if it exists else just the plain class name. This is so you can name the class OpenAIEmbeddingFunction and just look it up as registry.get("openai")

Reply:
ok got it